ecosystem/sep-0010.md

```diff
@@ -128,31 +129,37 @@ To validate the challenge transaction the following steps are performed by the s
+A server that needs to support validating accounts that do not exist can require a signature of the master key of the account address.
 
-Some servers may have use cases to issue JWTs for less than all thresholds.
+A server that needs to verify a user has complete signing authority of an account should verify that the weight of the client signatures meet all thresholds of the account. It should do this by checking that the sum of the weights is equal or greater to all the thresholds: low, mid and high. The server should only issue a JWT if all thresholds are met. When determining the weight of the client signatures the server's signature should be explicitly excluded as it may be a signer on the account and a server verifying authentication should not help the client authenticate.
 
-When determining the weight of the client signatures the server's signature should be explicitly excluded as it may be a signer on the account and a server verifying authentication should not help the client authenticate.
+A server may choose to issue JWTs for less than all thresholds and based on any other application specific logic.
 
 Upon successful validation service responds with a session JWT, containing the following claims:
 
 * `iss` (the principal that issued a token, [RFC7519, Section 4.1.1](https://tools.ietf.org/html/rfc7519#section-4.1.1)) — a [Uniform Resource Identifier (URI)] for the issuer (`https://example.com` or `https://example.com/G...`)
 * `sub` (the principal that is the subject of the JWT, [RFC7519, Section 4.1.2](https://tools.ietf.org/html/rfc7519#section-4.1.2)) — the public key of the authenticating Stellar account (`G...`)
 * `iat` (the time at which the JWT was issued [RFC7519, Section 4.1.6](https://tools.ietf.org/html/rfc7519#section-4.1.6)) — current timestamp (`1530644093`)
 * `exp` (the expiration time on or after which the JWT must not be accepted for processing, [RFC7519, Section 4.1.4](https://tools.ietf.org/html/rfc7519#section-4.1.4)) — a server can pick its own expiration period for the token (`1530730493`)
-* `jti` (the unique identifier for the JWT, [RFC7519, Section 4.1.7](https://tools.ietf.org/html/rfc7519#section-4.1.7)) — hex-encoded hash of the challenge transaction (`f0e754c6ab988eb5fb2811c2cacc4d3d2aa4334763b8df7285ef341921e2530a`)
+* `exs` — boolen indicating if the account exists
+* `sig` — array of public keys of the signers of the challenge transaction
 * `thr`
   * `low` — low threshold of the account
   * `med` — medium threshold of the account
   * `high` — high threshold of the account
 * `wht` — weight of the signer(s) that signed the challenge transaction
 
-The JWT may contain other claims specific to your application, see [RFC7519].
+The JWT may contain other claims specific to your application, see [RFC7519], or omit claims that will not be used in your scenarios.
```

Comment:
I might make two sections, one with the minimal set of claims for the JWT to function and one for suggested claims that may be useful in difference scenarios. At the end of the day the server is almost certainly making a JWT for its own use, so I wouldn't want implementers to get bogged down trying to implement many of these extra fields unless they've explicitly decided they need them.

Reply:
Thanks, that's a better way to present this. 685ffb9